app/repositories/CampaignContentRepository.scala

```diff
@@ -1,75 +1,63 @@
 
   def getContentForCampaign(campaignId: String): Either[CampaignCentralApiError, List[ContentItem]] = {
-    val result: List[Either[CampaignCentralApiError, ContentItem]] = getItemsForCampaignId(campaignId).map {
-      ContentItem.fromItem
-    }
-    result.sequence
-  }
-
-  def getContent(campaignId: String, id: String): Either[CampaignCentralApiError, ContentItem] = {
-    val contentItemOrError: Either[CampaignCentralApiError, Option[ContentItem]] = {
-      val result: Option[Either[CampaignCentralApiError, ContentItem]] =
-        Option(Dynamo.campaignContentTable.getItem("campaignId", campaignId, "id", id)).map { ContentItem.fromItem }
-      result.sequence
-    }
-
-    contentItemOrError.flatMap(
-      item =>
-        item.map(Right(_)) getOrElse Left(
-          ContentItemNotFound(s"Could not find content item with campaign id $campaignId and id $id")))
-  }
-
-  def deleteContentForCampaign(campaignId: String): Either[CampaignCentralApiError, DeleteCampaignContentResult] = {
-    Try[List[DeleteItemOutcome]] {
-      for (content <- getItemsForCampaignId(campaignId))
-        yield Dynamo.campaignContentTable.deleteItem("campaignId", campaignId, "id", content.getString("id"))
-    } match {
-      case Success(results) =>
-        Logger.info(s"Deleted ${results.length} items for $campaignId")
-        Right(DeleteCampaignContentResult(campaignId, results.length, results))
-      case Failure(exception) =>
-        Logger.error(s"failed to delete content for campaign $campaignId", exception)
-        Left(CampaignItemDeletionFailed(campaignId, exception))
+    val result = Scanamo.query[ContentItem](DynamoClient)(tableName)('campaignId -> campaignId)
+    getResultsOrFirstFailure(result).left map { e =>
+      JsonParsingError(e.show)
     }
   }
 
-  def putContent(contentItem: ContentItem): Either[CampaignCentralApiError, PutContentItemResult] = {
-    contentItem.toItem match {
-      case Left(error) => Left(error)
-      case Right(item) =>
-        Try(Dynamo.campaignContentTable.putItem(item)) match {
-          case Success(putItemOutcome) =>
-            Logger.info(s"Successfully put ContentItem $contentItem; ${putItemOutcome.toString}")
-            Right(PutContentItemResult(contentItem, putItemOutcome))
+  def getContent(campaignId: String, id: String): Either[CampaignCentralApiError, ContentItem] =
+    Scanamo.get[ContentItem](DynamoClient)(tableName)('campaignId -> campaignId and 'id -> id) map {
+      case Left(e)            => Left(JsonParsingError(e.show))
+      case Right(contentItem) => Right(contentItem)
+    } getOrElse
+      Left(ContentItemNotFound(s"Could not find content item with campaign id $campaignId and id $id"))
+
+  def deleteContentForCampaign(campaignId: String): Either[CampaignCentralApiError, DeleteCampaignContentResult] =
+    getContentForCampaign(campaignId) match {
+      case Left(e) => Left(e)
+      case Right(content) =>
+        Try {
```

Comment:
Why do you need a `Try` here?

Reply:
Going to guess before @kelvin-chappell answers... I think I remember @philwills saying that there are some exceptions coming from the underlaying dynamo client in Scanamo that aren't translated to types yet. That's my guess! 😄

Reply:
That's what I was trying to catch.  Not sure if it's necessary but I'll check.

Reply:
Yes, transport level and dynamo semantic errors are currently thrown, which is frankly a bit rubbish and something I'd love to change (PRs welcome 😀 ).